internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -696,10 +576,230 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
-func handleUpdateError(err error) (bool, ctrl.Result, error) {
-	if IsHTTP4XXError(err) {
-		return true, ctrl.Result{}, errors.Wrap(err, "failed to update clusterDeployment")
+// updateStatus is updating all the ClusterDeployment Conditions.
+// In case that an error has occured when trying to sync the Spec, the error (syncErr) is presented in SpecSyncedCondition.
+// Internal bool differentiate between backend server error (internal HTTP 5XX) and user input error (HTTP 4XXX)
+func (r *ClusterDeploymentsReconciler) updateStatus(ctx context.Context, cluster *hivev1.ClusterDeployment, c *common.Cluster, syncErr error, internal bool) (ctrl.Result, error) {
```

Comment:
The usage of this function is always the same, you pass an error and internal is always `!IsHTTP4XXError(err)` maybe this check can move into the function?

Reply:
Good idea!